tests/callbacks/test_gpu_stats_monitor.py

```diff
@@ -84,7 +87,11 @@ def test_gpu_stats_monitor_no_queries(tmpdir):
         devices=1,
         callbacks=[gpu_stats],
     )
-    with mock.patch("pytorch_lightning.loggers.tensorboard.TensorBoardLogger.log_metrics") as log_metrics_mock:
+    with mock.patch(
+        "pytorch_lightning.loggers.tensorboard.TensorBoardLogger.log_metrics"
+    ) as log_metrics_mock, pytest.deprecated_call(
+        match="Trainer.data_parallel_device_ids` was deprecated in v1.6 and will be removed in v1.8."
```

Comment:
https://github.com/PyTorchLightning/pytorch-lightning/pull/12072#discussion_r813308009

This was still getting triggered, I previously migrated this callsite as well. But since it's going to be removed in v1.6, I kept using `Trainer.data_parallel_device_ids`. Any suggestions?

Reply:
No, it's not going to be removed in 1.6. It's going to be removed in 1.8 which means until then, if you don't change anything, the user would see this warning and they can't do anything about it, because the logger connector triggers it internally. We can't have that. 

What's clear is that we still need  a way to get the device ids. So my suggestion is to keep the method in the connector, deprecate the one on the trainer and then call the connector directly.

Reply:
That make sense! I feel the primary purpose of this diff is to remove `Accelerator.parallel_device_ids` as it's not intended to be a public API and internal usages can be moved to derive from `Trainer.strategy.parallel_devices`. I would prefer still remove this property.

Maybe I can hold-off this PR until I introduce `Trainer.device_ids` in #12126? Then `Trainer.data_parallel_device_ids` will be an easy migration. 

Thoughts?

Reply:
> Maybe I can hold-off this PR until I introduce Trainer.device_ids in https://github.com/PyTorchLightning/pytorch-lightning/issues/12126? Then

Yes, this would be my preference.

Reply:
Yeah make sense, having trainer.device_ids is the desired state for 1.6, we can have that first then deprecate the rest
Thanks @awaelchli for suggesting. Thank you @DuYicong515 for working on this :)

Reply:
Hi @awaelchli, I just updated the PR, do you mind to recheck the changes here? Thanks!